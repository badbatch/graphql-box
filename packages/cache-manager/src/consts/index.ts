export const METADATA = "metadata";
export const CACHE_CONTROL = "cacheControl";
export const NO_CACHE = "noCache";

export const HEADER_CACHE_CONTROL = "cache-control";
export const HEADER_NO_CACHE = "no-cache";

export const CACHE_ENTRY_ADDED = "cache_entry_added";
export const CACHE_ENTRY_QUERIED = "cache_entry_queried";
export const PARTIAL_QUERY_COMPILED = "partial_query_compiled";
