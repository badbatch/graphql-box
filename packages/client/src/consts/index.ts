export const REQUEST_EXECUTED = "request_executed";
export const SUBSCRIPTION_EXECUTED = "subscription_executed";
