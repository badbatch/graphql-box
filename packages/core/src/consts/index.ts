export const DEFAULT_TYPE_ID_KEY = "id";
export const TYPE_NAME_KEY = "__typename";

export const MUTATION = "mutation";
export const QUERY = "query";
export const SUBSCRIPTION = "subscription";

export const DATA_ENTITIES = "dataEntities";
export const QUERY_RESPONSES = "queryResponses";
export const REQUEST_FIELD_PATHS = "requestFieldPaths";
