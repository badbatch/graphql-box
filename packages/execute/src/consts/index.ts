export const EXECUTE_EXECUTED = "execute_executed";
