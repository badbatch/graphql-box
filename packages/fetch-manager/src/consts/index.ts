export const FETCH_EXECUTED = "fetch_executed";
