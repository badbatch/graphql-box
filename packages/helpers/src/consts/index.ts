export const DOCUMENT = "Document";
export const FIELD = "Field";
export const FRAGMENT_DEFINITION = "FragmentDefinition";
export const FRAGMENT_SPREAD = "FragmentSpread";
export const INLINE_FRAGMENT = "InlineFragment";
export const LIST_VALUE = "ListValue";
export const NAME = "name";
export const OBJECT_VALUE = "ObjectValue";
export const OF_TYPE = "ofType";
export const OPERATION_DEFINITION = "OperationDefinition";
export const STRING = "String";
export const TYPE = "type";
export const VALUE = "value";
export const VARIABLE = "Variable";
export const VARIABLE_DEFINITION = "VariableDefinition";
