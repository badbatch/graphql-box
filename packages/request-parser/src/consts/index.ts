export const DOCUMENT = "Document";
export const FIELD = "Field";
export const FRAGMENT_DEFINITION = "FragmentDefinition";
export const FRAGMENT_SPREAD = "FragmentSpread";
export const INLINE_FRAGMENT = "InlineFragment";
export const LIST_VALUE = "ListValue";
export const NAME = "name";
export const OBJECT_VALUE = "ObjectValue";
export const OF_TYPE = "ofType";
export const OPERATION_DEFINITION = "OperationDefinition";
export const VALUE = "value";
export const VARIABLE = "Variable";

export const QUERY = "query";
export const METADATA = "_metadata";

export const DEFAULT_TYPE_ID_KEY = "id";
