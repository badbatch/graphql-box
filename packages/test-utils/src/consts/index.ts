export const EMAIL_ADDED = "emailAdded";
export const EMAILS_DELETED = "emailsDeleted";
