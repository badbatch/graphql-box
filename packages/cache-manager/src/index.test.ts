import Core from "@cachemap/core";
import map from "@cachemap/map";
import { RequestData, ResponseData } from "@handl/core";
import {
  getRequestContext,
  getRequestData,
  githubParsedQueries,
  githubQueryFieldTypeMaps,
  githubQueryResponses,
} from "@handl/test-utils";
import { CacheManager, CacheManagerDef } from ".";
import { DEFAULT_TYPE_ID_KEY } from "./consts";

describe("@handl/cache-manager", () => {
  let cacheManager: CacheManagerDef;

  describe("the resolveQuery method", () => {
    let responseData: ResponseData;
    let requestData: RequestData;

    describe("when caching is done through cascading cache control", () => {
      beforeAll(async () => {
        cacheManager = await CacheManager.init({
          cache: await Core.init({
            name: "cachemap",
            store: map(),
          }),
          cascadeCacheControl: true,
          typeIDKey: DEFAULT_TYPE_ID_KEY,
        });

        requestData = getRequestData(githubParsedQueries.simple);

        responseData = await cacheManager.resolveQuery(
          requestData,
          requestData,
          githubQueryResponses.simple,
          { awaitDataCaching: true },
          getRequestContext({ fieldTypeMap: githubQueryFieldTypeMaps.simple }),
        );
      });

      it("then the method should return the correct response data", () => {
        expect(responseData).toMatchSnapshot();
      });

      it("then the cache should contain the correct data", async () => {
        expect(await cacheManager.cache.export()).toMatchSnapshot();
      });
    });

    describe("when caching is done through type cache directives", () => {
      beforeAll(async () => {
        cacheManager = await CacheManager.init({
          cache: await Core.init({
            name: "cachemap",
            store: map(),
          }),
          typeCacheDirectives: {
            Organization: "public, max-age=1",
          },
          typeIDKey: DEFAULT_TYPE_ID_KEY,
        });

        requestData = getRequestData(githubParsedQueries.simple);

        responseData = await cacheManager.resolveQuery(
          requestData,
          requestData,
          githubQueryResponses.simple,
          { awaitDataCaching: true },
          getRequestContext({ fieldTypeMap: githubQueryFieldTypeMaps.simple }),
        );
      });

      it("then the method should return the correct response data", () => {
        expect(responseData).toMatchSnapshot();
      });

      it("then the cache should contain the correct data", async () => {
        expect(await cacheManager.cache.export()).toMatchSnapshot();
      });
    });
  });
});
