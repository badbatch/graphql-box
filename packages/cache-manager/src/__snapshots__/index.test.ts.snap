// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@handl/cache-manager the resolveQuery method when caching is done through cascading cache control then the cache should contain the correct data 1`] = `
Object {
  "entries": Array [
    Array [
      "requestFieldPaths::5256db516a2438a7e5488a00413f4809",
      Object {
        "description": "We are working to build community through open source technology. NB: members must have two-factor auth.",
        "email": "",
        "id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
        "login": "facebook",
        "name": "Facebook",
        "url": "https://github.com/facebook",
      },
    ],
    Array [
      "dataEntities::Organization:MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
      Object {
        "description": "We are working to build community through open source technology. NB: members must have two-factor auth.",
        "email": "",
        "id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
        "login": "facebook",
        "name": "Facebook",
        "url": "https://github.com/facebook",
      },
    ],
    Array [
      "queryResponses::651f14b3df12bef942ed3b1ba9e7f1f1",
      Object {
        "cacheMetadata": Object {
          "query": Object {
            "cacheControl": Object {
              "maxAge": 5,
              "public": true,
            },
            "etag": undefined,
            "ttl": 1550528399907,
          },
          "query.organization": Object {
            "cacheControl": Object {
              "maxAge": 5,
              "public": true,
            },
            "etag": undefined,
            "ttl": 1550528399907,
          },
        },
        "data": Object {
          "organization": Object {
            "description": "We are working to build community through open source technology. NB: members must have two-factor auth.",
            "email": "",
            "id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
            "login": "facebook",
            "name": "Facebook",
            "url": "https://github.com/facebook",
          },
        },
      },
    ],
  ],
  "metadata": Array [
    Object {
      "accessedCount": 0,
      "added": 1550528394914,
      "cacheability": Cacheability {
        "metadata": Object {
          "cacheControl": Object {
            "maxAge": 5,
            "public": true,
          },
          "etag": undefined,
          "ttl": 1550528399914,
        },
      },
      "key": "dataEntities::Organization:MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
      "lastAccessed": 1550528394914,
      "lastUpdated": 1550528394914,
      "size": 410,
      "tags": Array [],
      "updatedCount": 0,
    },
    Object {
      "accessedCount": 0,
      "added": 1550528394914,
      "cacheability": Cacheability {
        "metadata": Object {
          "cacheControl": Object {
            "maxAge": 5,
            "public": true,
          },
          "etag": undefined,
          "ttl": 1550528399914,
        },
      },
      "key": "queryResponses::651f14b3df12bef942ed3b1ba9e7f1f1",
      "lastAccessed": 1550528394914,
      "lastUpdated": 1550528394914,
      "size": 678,
      "tags": Array [],
      "updatedCount": 0,
    },
    Object {
      "accessedCount": 0,
      "added": 1550528394913,
      "cacheability": Cacheability {
        "metadata": Object {
          "cacheControl": Object {
            "maxAge": 5,
            "public": true,
          },
          "etag": undefined,
          "ttl": 1550528399913,
        },
      },
      "key": "requestFieldPaths::5256db516a2438a7e5488a00413f4809",
      "lastAccessed": 1550528394913,
      "lastUpdated": 1550528394913,
      "size": 410,
      "tags": Array [],
      "updatedCount": 0,
    },
  ],
}
`;

exports[`@handl/cache-manager the resolveQuery method when caching is done through cascading cache control then the method should return the correct response data 1`] = `
Object {
  "cacheMetadata": Map {
    "query" => Cacheability {
      "metadata": Object {
        "cacheControl": Object {
          "maxAge": 5,
          "public": true,
        },
        "etag": undefined,
        "ttl": 1550528399907,
      },
    },
    "query.organization" => Cacheability {
      "metadata": Object {
        "cacheControl": Object {
          "maxAge": 5,
          "public": true,
        },
        "etag": undefined,
        "ttl": 1550528399907,
      },
    },
  },
  "data": Object {
    "organization": Object {
      "description": "We are working to build community through open source technology. NB: members must have two-factor auth.",
      "email": "",
      "id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
      "login": "facebook",
      "name": "Facebook",
      "url": "https://github.com/facebook",
    },
  },
}
`;

exports[`@handl/cache-manager the resolveQuery method when caching is done through type cache directives then the cache should contain the correct data 1`] = `
Object {
  "entries": Array [
    Array [
      "requestFieldPaths::5256db516a2438a7e5488a00413f4809",
      Object {
        "description": "We are working to build community through open source technology. NB: members must have two-factor auth.",
        "email": "",
        "id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
        "login": "facebook",
        "name": "Facebook",
        "url": "https://github.com/facebook",
      },
    ],
    Array [
      "dataEntities::Organization:MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
      Object {
        "description": "We are working to build community through open source technology. NB: members must have two-factor auth.",
        "email": "",
        "id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
        "login": "facebook",
        "name": "Facebook",
        "url": "https://github.com/facebook",
      },
    ],
    Array [
      "queryResponses::651f14b3df12bef942ed3b1ba9e7f1f1",
      Object {
        "cacheMetadata": Object {
          "query": Object {
            "cacheControl": Object {
              "maxAge": 1,
              "public": true,
            },
            "etag": undefined,
            "ttl": 1550528395921,
          },
          "query.organization": Object {
            "cacheControl": Object {
              "maxAge": 1,
              "public": true,
            },
            "etag": undefined,
            "ttl": 1550528395921,
          },
        },
        "data": Object {
          "organization": Object {
            "description": "We are working to build community through open source technology. NB: members must have two-factor auth.",
            "email": "",
            "id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
            "login": "facebook",
            "name": "Facebook",
            "url": "https://github.com/facebook",
          },
        },
      },
    ],
  ],
  "metadata": Array [
    Object {
      "accessedCount": 0,
      "added": 1550528394921,
      "cacheability": Cacheability {
        "metadata": Object {
          "cacheControl": Object {
            "maxAge": 1,
            "public": true,
          },
          "etag": undefined,
          "ttl": 1550528395921,
        },
      },
      "key": "requestFieldPaths::5256db516a2438a7e5488a00413f4809",
      "lastAccessed": 1550528394921,
      "lastUpdated": 1550528394921,
      "size": 410,
      "tags": Array [],
      "updatedCount": 0,
    },
    Object {
      "accessedCount": 0,
      "added": 1550528394921,
      "cacheability": Cacheability {
        "metadata": Object {
          "cacheControl": Object {
            "maxAge": 1,
            "public": true,
          },
          "etag": undefined,
          "ttl": 1550528395921,
        },
      },
      "key": "dataEntities::Organization:MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
      "lastAccessed": 1550528394921,
      "lastUpdated": 1550528394921,
      "size": 410,
      "tags": Array [],
      "updatedCount": 0,
    },
    Object {
      "accessedCount": 0,
      "added": 1550528394921,
      "cacheability": Cacheability {
        "metadata": Object {
          "cacheControl": Object {
            "maxAge": 1,
            "public": true,
          },
          "etag": undefined,
          "ttl": 1550528395921,
        },
      },
      "key": "queryResponses::651f14b3df12bef942ed3b1ba9e7f1f1",
      "lastAccessed": 1550528394921,
      "lastUpdated": 1550528394921,
      "size": 678,
      "tags": Array [],
      "updatedCount": 0,
    },
  ],
}
`;

exports[`@handl/cache-manager the resolveQuery method when caching is done through type cache directives then the method should return the correct response data 1`] = `
Object {
  "cacheMetadata": Map {
    "query" => Cacheability {
      "metadata": Object {
        "cacheControl": Object {
          "maxAge": 1,
          "public": true,
        },
        "etag": undefined,
        "ttl": 1550528395921,
      },
    },
    "query.organization" => Cacheability {
      "metadata": Object {
        "cacheControl": Object {
          "maxAge": 1,
          "public": true,
        },
        "etag": undefined,
        "ttl": 1550528395921,
      },
    },
  },
  "data": Object {
    "organization": Object {
      "description": "We are working to build community through open source technology. NB: members must have two-factor auth.",
      "email": "",
      "id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
      "login": "facebook",
      "name": "Facebook",
      "url": "https://github.com/facebook",
    },
  },
}
`;
